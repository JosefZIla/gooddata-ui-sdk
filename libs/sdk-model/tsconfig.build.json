{
    "extends": "./tsconfig.json",
    "include": ["src/index.ts", "src/dashboard/drillUrl.ts"],
    "compilerOptions": {
        "baseUrl": null,
        "paths": null,
        "declaration": true,
        "declarationMap": true
    }
}
