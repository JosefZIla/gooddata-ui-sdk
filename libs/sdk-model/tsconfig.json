{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es5",
        "module": "commonjs",
        "moduleResolution": "node",
        "importHelpers": true,
        "lib": ["es6", "dom", "ES2016.Array.Include"],
        "declaration": true,
        "declarationMap": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "removeComments": false
    },
    "exclude": ["**/node_modules", "dist", "esm"]
}
