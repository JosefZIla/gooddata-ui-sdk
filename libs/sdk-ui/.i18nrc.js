module.exports = {
    paths: ["./src/base/localization/bundles"],
    structure: true,
    intl: true,
    html: true,
    insightToReport: true,
    usage: false,
    debug: false,
    source: "src/**/*.{ts,js,tsx,jsx}",
    rules: [
        {
            pattern: [/.+/],
        },
    ],
};
