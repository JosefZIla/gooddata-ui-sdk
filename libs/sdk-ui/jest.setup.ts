// (C) 2019 GoodData Corporation
/* eslint-disable import/no-unassigned-import */
import * as raf from "raf";
import "@testing-library/jest-dom/extend-expect";

raf.polyfill();
