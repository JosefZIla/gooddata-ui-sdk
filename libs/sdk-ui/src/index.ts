// (C) 2007-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
import "./polyfills";
// new exports

export * from "./base";
export * from "./execution";
export * from "./kpi";

/**
 * Common interface uses to specify number separators for the different SDK components.
 * @public
 */
export { ISeparators } from "@gooddata/sdk-backend-spi";
