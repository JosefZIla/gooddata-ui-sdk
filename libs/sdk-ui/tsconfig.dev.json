{
    "extends": "./tsconfig.build.esm.json",
    "compilerOptions": {
        "noImplicitAny": false,
        "noUnusedLocals": false,
        "noUnusedParameters": false,
        "pretty": true
    }
}
