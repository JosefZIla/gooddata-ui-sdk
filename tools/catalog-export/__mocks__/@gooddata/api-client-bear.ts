// (C) 2007-2018 GoodData Corporation
import catalogItemsResponse from "./loadItems.json";
import getObjectsResponse from "./getObjects.json";
import getObjectsByQueryResponse from "./getObjectsByQuery.json";
import getVisualizationsResponse from "./getVisualizations.json";
import getAnalyticalDashboards from "./getAnalyticalDashboards.json";

export default {
    catalogue: {
        loadItems() {
            return Promise.resolve(catalogItemsResponse);
        },
    },
    md: {
        getObjectsByQuery() {
            return Promise.resolve(getObjectsByQueryResponse);
        },
        getObjects() {
            return Promise.resolve(getObjectsResponse);
        },
        getVisualizations() {
            return Promise.resolve(getVisualizationsResponse);
        },
        getAnalyticalDashboards() {
            return Promise.resolve(getAnalyticalDashboards);
        },
    },
};
