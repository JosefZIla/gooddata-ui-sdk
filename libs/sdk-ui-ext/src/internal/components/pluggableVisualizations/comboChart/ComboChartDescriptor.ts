// (C) 2021-2022 GoodData Corporation
import { IComboChartProps } from "@gooddata/sdk-ui-charts";
import { BucketNames } from "@gooddata/sdk-ui";

import {
    IVisualizationDescriptor,
    PluggableVisualizationFactory,
} from "../../../interfaces/VisualizationDescriptor";
import { PluggableComboChart } from "./PluggableComboChart";
import { BigChartDescriptor } from "../BigChartDescriptor";
import {
    getReactEmbeddingCodeGenerator,
    getInsightToPropsConverter,
    chartAdditionalFactories,
    filtersInsightConversion,
    sortsInsightConversion,
    chartConfigInsightConversion,
    multipleMeasuresBucketConversion,
    multipleAttributesBucketConversion,
} from "../../../utils/embeddingCodeGenerator";

export class ComboChartDescriptor extends BigChartDescriptor implements IVisualizationDescriptor {
    public getFactory(): PluggableVisualizationFactory {
        return (params) => new PluggableComboChart(params);
    }

    public getEmbeddingCode = getReactEmbeddingCodeGenerator({
        component: {
            importType: "named",
            name: "ComboChart",
            package: "@gooddata/sdk-ui-charts",
        },
        insightToProps: getInsightToPropsConverter<IComboChartProps>({
            primaryMeasures: multipleMeasuresBucketConversion("primaryMeasures", BucketNames.MEASURES),
            secondaryMeasures: multipleMeasuresBucketConversion(
                "secondaryMeasures",
                BucketNames.SECONDARY_MEASURES,
            ),
            viewBy: multipleAttributesBucketConversion("viewBy", BucketNames.VIEW),
            filters: filtersInsightConversion("filters"),
            sortBy: sortsInsightConversion("sortBy"),
            config: chartConfigInsightConversion("config"),
        }),
        additionalFactories: chartAdditionalFactories,
    });
}
