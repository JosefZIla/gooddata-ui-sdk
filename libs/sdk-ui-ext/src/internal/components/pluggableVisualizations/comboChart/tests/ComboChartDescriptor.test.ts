// (C) 2022 GoodData Corporation
import { ReferenceMd } from "@gooddata/reference-workspace";
import { modifyAttribute, modifyMeasure, newBucket, newInsightDefinition } from "@gooddata/sdk-model";
import { BucketNames } from "@gooddata/sdk-ui";
import { ComboChartDescriptor } from "../ComboChartDescriptor";

describe("ComboChartDescriptor", () => {
    it("should generate embedding code without context", () => {
        // TODO replace by some recording?
        const insight = newInsightDefinition("local:combo", (i) =>
            i.buckets([
                newBucket(
                    BucketNames.MEASURES,
                    modifyMeasure(ReferenceMd.Amount, (m) =>
                        m.alias("Custom Amount alias").localId("amount-custom-local-id"),
                    ),
                ),
                newBucket(
                    BucketNames.SECONDARY_MEASURES,
                    modifyMeasure(ReferenceMd.Velocity.Sum, (m) =>
                        m.alias("Custom Velocity alias").localId("velocity-custom-local-id"),
                    ),
                ),
                newBucket(
                    BucketNames.VIEW,
                    modifyAttribute(ReferenceMd.Region, (a) =>
                        a.alias("Custom Region alias").localId("region-custom-local-id"),
                    ),
                ),
            ]),
        );
        const descriptor = new ComboChartDescriptor();

        expect(descriptor.getEmbeddingCode(insight)).toMatchSnapshot();
    });
});
