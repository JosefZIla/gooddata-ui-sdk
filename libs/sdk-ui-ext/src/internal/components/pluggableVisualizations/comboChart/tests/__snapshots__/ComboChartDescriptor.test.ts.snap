// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboChartDescriptor should generate embedding code without context 1`] = `
"import React from \\"react\\";
import { idRef, newAttribute, newMeasure } from \\"@gooddata/sdk-model\\";
import { ComboChart } from \\"@gooddata/sdk-ui-charts\\";

const primaryMeasures = [
    newMeasure(idRef(\\"aangOxLSeztu\\", \\"measure\\"), m => m.alias(\\"Custom Amount alias\\").localId(\\"m_aangOxLSeztu\\"))
];
const secondaryMeasures = [
    newMeasure(idRef(\\"fact.stagehistory.velocity\\", \\"fact\\"), m => m.alias(\\"Custom Velocity alias\\").localId(\\"m_fact.stagehistory.velocity_sum\\").aggregation(\\"sum\\"))
];
const viewBy = [
    newAttribute(idRef(\\"label.owner.region\\", \\"displayForm\\"), a => a.alias(\\"Custom Region alias\\").localId(\\"a_label.owner.region\\"))
];
const style = { height: 400 };

function MyComponent() {
    return (
        <div style={style}>
            <ComboChart
                primaryMeasures={primaryMeasures}
                secondaryMeasures={secondaryMeasures}
                viewBy={viewBy}
            />
        </div>
    );
}
"
`;
